### PR TITLE
Add kubectl gs FAQ and Troubleshooting page

### PR DESCRIPTION
**Pull request by marians**
This PR adds two new pages:

- A page under https://docs.giantswarm.io/ui-api/gsctl/ about migrating from gsctl to kubectl-gs
- A page under https://docs.giantswarm.io/ui-api/kubectl-gs/ for FAQ and troubleshooting around kubectl gs

### CODE REVIEW

File: src/content/ui-api/gsctl/migrate.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+linkTitle: Migration to kubectl gs
+title: Migrating from gsctl to kubectl gs
+description: gsctl is a CLI for the deprecated Giant Swarm Rest API. As you move from the Rest API towards the Management API, you'll have to say goodby to gsctl and embrace kubectl-gs, step by step. This page should help you make the transition smoothly.
```
Comment (axbarsan) [inline src/content/ui-api/gsctl/migrate.md:4]:
```suggestion
description: gsctl is a CLI for the deprecated Giant Swarm REST API. As you move from the REST API towards the Management API, you'll have to say good bye to gsctl and embrace kubectl-gs, step by step. This page should help you make the transition smoothly.
```

File: src/content/ui-api/gsctl/migrate.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+weight: 1
+menu:
+  main:
+    parent: uiapi-gsctl
+user_questions:
+- What commands in kubectl-gs replace what gsctl commands?
+- How can I replace gsctl with kubectl-gs?
```
Comment (axbarsan) [inline src/content/ui-api/gsctl/migrate.md:11]: Should we call it `kubectl gs` or `kubectl-gs`?

File: src/content/ui-api/gsctl/migrate.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-rainbow
+last_review_date: 2021-11-23
+---
+
+# Migrating from `gsctl` to `kubectl gs`
+
+`gsctl` is a CLI for Giant Swarm's proprietary Rest API, which is no longer developed. To make full use of clusters and apps with Giant Swarm, users have to migrate to the [Management API]({{< relref "/ui-api/management-api/" >}}).
```
Comment (axbarsan) [inline src/content/ui-api/gsctl/migrate.md:19]:
```suggestion
`gsctl` is a CLI for Giant Swarm's proprietary REST API, which is no longer developed. To make full use of clusters and apps with Giant Swarm, users have to migrate to the [Management API]({{< relref "/ui-api/management-api/" >}}).
```

File: src/content/ui-api/gsctl/migrate.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+| gsctl show cluster      | [kubectl gs get clusters]({{< relref "/ui-api/kubectl-gs/get-clusters.md" >}}) |
+| gsctl show nodepool     | [kubectl gs get nodepools]({{< relref "/ui-api/kubectl-gs/get-nodepools.md" >}}) |
+| gsctl show release      | [kubectl gs get releases]({{< relref "/ui-api/kubectl-gs/get-releases.md" >}}) |
+
+Below is some additional information on commands that don't have a direct replacement.
+
+| gsctl command | Comment |
+|---------------|---------|
+| gsctl completion | `kubectl-gs` does not yet provide support for shell completion. |
+| gsctl delete cluster | To delete a cluster, use `kubectl delete` on the cluster's main resource. |
```
Comment (axbarsan) [inline src/content/ui-api/gsctl/migrate.md:46]: AFAIK on AWS you have to delete each resource manually

File: src/content/ui-api/gsctl/migrate.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+| gsctl delete nodepool | To delete a node pool, use `kubectl delete` on the node pool's main resource. |
```
Comment (axbarsan) [inline src/content/ui-api/gsctl/migrate.md:47]: Same here

File: src/content/ui-api/kubectl-gs/faq.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+You can simply ask your Giant Swarm support contact. As an alternative, `kubectl gs login` also accepts the web UI URL as an argument. So if you happen to know that one, simply use that. (The logic behind this is quite simple: The domain prefix `happa.` simply gets removed from the web UI URL, and the result is the Management API endpoint URL.)
+
+### What does `Error: Token renewal failed` mean
+
+It means that `kubectl gs login` tried to get a fresh ID-token to authenticate with against the Kubernetes API, since the existing one had expired. To resolve this, please check your `kubectl` configuration file (typically in `~/.kube/config`). Do you have several `user` entries for the same management cluster, using the same `client-id`? If yes, please edit your configuration so that there is only one.
+
+### Why does the command connect to a service called `athena`
+
+The Management API uses a TLS certificate signed by a custom certificate authority (CA). In order to communicate with the API, a client (here: `kubectl`) must trust that CA, it must have the CA's certificate. Athena is a public service providing that CA certificate to the client.
```
Comment (axbarsan) [inline src/content/ui-api/kubectl-gs/faq.md:53]: It also provides installation metadata (provider and name), the K8s API URL and the authentication provider's URL

File: src/content/ui-api/gsctl/migrate.md (hunk 2 above)
Comment (marians) [reply src/content/ui-api/gsctl/migrate.md:11]: I settled for kubectl-gs when it comes to the tool's name and `kubectl gs ...` only when it's about invoking a command.

File: src/content/ui-api/gsctl/migrate.md (hunk 4 above)
Comment (marians) [reply src/content/ui-api/gsctl/migrate.md]: Changed the wording so it matches all cases.

Can't really go into the details in this places. We seem to need a `delete` command. See https://github.com/giantswarm/roadmap/issues/409

File: src/content/ui-api/kubectl-gs/faq.md (hunk 6 above)
Comment (marians) [reply src/content/ui-api/kubectl-gs/faq.md]: Added a bit more here.

**Comment by marians** [comment]
I think I processed all input. Thanks!

File: src/content/ui-api/gsctl/migrate.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
+---
+linkTitle: Migration to kubectl gs
+title: Migrating from gsctl to kubectl gs
```
Comment (axbarsan) [inline src/content/ui-api/gsctl/migrate.md:3]: Should we have `kubectl-gs` or `kubectl gs` here?

File: src/content/ui-api/gsctl/migrate.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    parent: uiapi-gsctl
+user_questions:
+- What commands in kubectl-gs replace what gsctl commands?
+- How can I replace gsctl with kubectl-gs?
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-rainbow
+last_review_date: 2021-11-23
+---
+
+# Migrating from `gsctl` to `kubectl gs`
```
Comment (axbarsan) [inline src/content/ui-api/gsctl/migrate.md:17]: Same here

Comment (marians) [reply src/content/ui-api/gsctl/migrate.md]: Done

File: src/content/ui-api/gsctl/migrate.md (hunk 7 above)
Comment (marians) [reply src/content/ui-api/gsctl/migrate.md]: Done

**Review by axbarsan — APPROVED** [review]

**Pull request merged**